### PR TITLE
Relplat 897 copywrite fixes for mutliple licenses

### PR DESCRIPTION
**Pull request by modrake**
Creates a workaround for the copywrite bot that allows header checking on MPL licensed subdirs. Includes a per-subdir config file for each exception, and a new make target for executing copywrite bot runs. I've included a GHA workflow for the Vault team to integrate into the existing build process; example execution of this workflow can be seen under the "Check Copywrite Headers" workflow in Github Actions.

This PR also includes updated headers for all files found missing headers at the time of its development; this prevents a large delta for the first person who runs this workflow after it is implemented. I have added the ui header changes in their own commit and will bring in the ui team to test and ensure the new headers do not impact their templates in unexpected ways.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -316,3 +316,12 @@ ci-get-version-package:
 .PHONY: ci-prepare-legal
 ci-prepare-legal:
 	@$(CURDIR)/scripts/ci-helper.sh prepare-legal
+
+.PHONY: ci-copywriteheaders
+ci-copywriteheaders:
+	copywrite headers --plan
```
[the remark this answers is not in the document]
Comment (hghaf099) [reply Makefile:322]: It could also be postponed for later as that is our problem :)

File: Makefile [hunk 2]
```diff
@@ -316,3 +316,12 @@ ci-get-version-package:
...
+	# Special case for MPL headers in /api, /sdk, and /shamir
+	cd api && $(CURDIR)/scripts/copywrite-exceptions.sh
+	cd sdk && $(CURDIR)/scripts/copywrite-exceptions.sh
+	cd shamir && $(CURDIR)/scripts/copywrite-exceptions.sh
```
Comment (hghaf099) [inline Makefile:326]: It seems that `copywrite-excetions.sh` does indeed apply the changes. This is different that what the one in line 322 does. Should they be both do the same thing?

**Review by hghaf099 — COMMENTED** [review]
Thanks for helping with this! I just have a couple of nit questions, otherwise looks great!

File: Makefile (hunk 2 above)
Comment (modrake) [reply Makefile:326]: Thanks for catching this, it was an accident, I've updated all copywrite invocations to use the --plan flag.

**Review by hghaf099 — APPROVED** [review]
LGTM!

**Pull request merged**